pandas/core/groupby/ops.py

```diff
@@ -258,7 +258,10 @@ def groups(self):
         if len(self.groupings) == 1:
             return self.groupings[0].groups
         else:
-            to_groupby = zip(*(ping.grouper for ping in self.groupings))
+            to_groupby = zip(*(ping.grouper if not isinstance(ping.grouper,
+                                                              self.__class__)
+                             else ping.grouper.groupings[0].grouper for ping
```

Comment:
Wouldn't this only work if the grouper was the first item? Would need something more generalizable

Reply:
I was wondering the same. Can Groupers be nested to more than one level?

Reply:
I think this would fail if your test was `.groupby(['beta', pd.Grouper(level='alpha')])` instead of `.groupby([pd.Grouper(level='alpha'), 'beta'])`

Reply:
The problem is with the construction of the Groupings. The [0] indexes the sole BaseGrouper created inside the grouping due to the _get_grouper method. The order will be maintained. I have added another assertion to check this and it passed.